nemo/src/program_analysis/analysis.rs

```diff
@@ -660,6 +660,92 @@ impl ChaseProgram {
                 })
                 .collect::<HashMap<_, _>>();
 
+            let mut fact_and_constant_term_decls: HashMap<Identifier, Vec<Option<PrimitiveType>>> =
```

Comment:
This whole function was already very long and hard to follow. It may even deserve its own module. Would you think it is reasonable to split this thing off into a new module and break it into comprehensible pieces?